File: festival-web/src/main/java/cn/bdqfork/web/route/message/HttpMessageHandler.java
package cn.bdqfork.web.route.message;

import io.vertx.reactivex.ext.web.RoutingContext;

import java.lang.reflect.Parameter;

/**
 * @author bdq
 * @since 2020/1/31
 */
public interface HttpMessageHandler {
    Object[] handle(RoutingContext routingContext, Parameter[] parameters) throws Exception;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/message/HttpMessageHandler.kt
package cn.bdqfork.kotlin.web.route.message

import io.vertx.ext.web.RoutingContext
import java.lang.reflect.Parameter

/**
 * @author bdq
 * @since 2020/1/31
 */
interface HttpMessageHandler {
    @Throws(Exception::class)
    fun handle(routingContext: RoutingContext, parameters: Array<Parameter>): Array<Any?>
}